src/fortranscanner.l

```diff
@@ -899,8 +914,23 @@ private                                 {
                                           /* work around for bug in QCString.replace (QCString works) */
                                           QCString name=yytext;
                                           name = name.lower();
+                                          /* if variable/type/etc is part of a module, mod name is necessary for output */
+                                          // get surrounding state
+                                          int currentState = YY_START;
+                                          yy_pop_state(yyscanner);
+                                          int outerState = YY_START;
+                                          yy_push_state(currentState,yyscanner);
+                                          if( outerState == Start || outerState == ModuleBody )
+                                          {
+                                            if ((yyextra->current_root) && 
+                                                (yyextra->current_root->section == Entry::CLASS_SEC
+                                                 || yyextra->current_root->section == Entry::NAMESPACE_SEC))
+                                            {
+                                              name = yyextra->current_root->name + "::" + name;
```

Comment:
Is it guaranteed that `yyextra->current_root->name` is lowercase otherwise it is might be better to "lower()" it as well.

Reply:
I put a `name.lower().str()` in the line below - this seems more consistent with the rest of the code.